src/data_structures/fenwick_tree.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/**
+ * Fenwick Tree (Binary Indexed Tree)
+ */
+function FenwickTree(length) {
+  this._elements = new Array(length + 1);
+  for (var i = 0; i < this._elements.length ; i++)
+    this._elements[i] = 0;
+}
+
+/**
+ * Adds value to the array at specified index in O(log n)
+ */
+FenwickTree.prototype.adjust = function (index, value) {
+  for (; index < this._elements.length ; index += (index&-index))
```

Comment:
This loop (as well as the one in prefixSum is not really trivial to understand if you don't go read the algorithm specs. Would you mind adding some comments saying what they're doing?